### PR TITLE
Allow initialisation with State object

### PR DESCRIPTION
**Pull request by ferdinandsalis**
Possible implementation of https://github.com/MicheleBertoli/react-automata/issues/30

🙇‍♂️

### CODE REVIEW

**Comment by MicheleBertoli** [comment]
Thank you very much, @ferdinandsalis.
This is perfectly fine (except I'd try to avoid the nested ternary :)).

One question: what if we always assume that the `initialState`is a State object?
We could remove the `State.from` call and use the value provided to the component as it is.
If I'm not wrong the use case is saving the machine state (e.g. on exit) and reusing it, so that it should always be a valid state.
What do you think?

**Comment by ferdinandsalis** [comment]
@MicheleBertoli I have ammended the commit with the proposed changes.

🖖

**Comment by ferdinandsalis** [comment]
As a note I was initially off put by this line the xstate docs …

> new State(value, history?, actions?)
> Instantiates a new State instance. Typically, you will seldom need to create a new State instance, except possibly when testing.

Though this seems like a valid use case.

**Comment by davidkpiano** [comment]
@ferdinandsalis You're using it for testing, so it's 💯 % valid, no worries

File: src/withStatechart.js [hunk 1]
```diff
@@ -11,13 +11,13 @@ const withStatechart = (statechart, options = {}) => Component => {
     constructor(props) {
       super(props)
 
-      const initialMachineState = this.props.initialMachineState
-        ? State.from(this.props.initialMachineState)
+      const machineState = this.props.initialMachineState
```
Comment (MicheleBertoli) [inline src/withStatechart.js]: nit: `const machineState = this.props.initialMachineState || this.machine.initialState`

**Comment by MicheleBertoli** [comment]
Thank you very much, @ferdinandsalis @davidkpiano.

**Pull request merged**